matchbox-server/src/main/java/ch/ahdis/matchbox/validation/ValidationProvider.java

```diff
@@ -293,8 +294,21 @@ private IBaseResource getOperationOutcome(final String id,
 			issue.setDetails(null);
 
 			// Add slice info to diagnostics
-			if (message.sliceText != null) {
-				List<String> sliceInfo = engine.filterSlicingMessages(message.sliceText);
+			if (message.hasSliceInfo() && message.sliceHtml != null) {
+
+				List<String> liElements = new ArrayList<>();
+				String html = message.sliceHtml;
+				int startIndex = 0;
+				while ((startIndex = html.indexOf("<li>", startIndex)) != -1) {
+					int endIndex = html.indexOf("</li>", startIndex);
+					if (endIndex != -1) {
+						String content = html.substring(startIndex + 4, endIndex).trim();
+						liElements.add(content);
+						startIndex = endIndex + 5;
+					}
+				}
+
+				List<String> sliceInfo = engine.filterSlicingMessages(liElements.toArray(new String[0]));
```

Comment:
We may move the list item splitting logic into `engine.filterSlicingMessages`, since it's duplicated in `GazelleValidationWs`.

Reply:
thanks did that